docs/site/Interceptors.md

```diff
@@ -0,0 +1,255 @@
+  console.log('convertName: before-' + invocationCtx.methodName);
+  invocationCtx.args[0] = (invocationCtx.args[0] as string).toUpperCase();
+  const result = await next();
+  console.log('convertName: after-' + invocationCtx.methodName);
+  return result;
+};
+```
+
+## Apply interceptors
+
+Interceptors form a cascading chain of handlers around the target method
+invocation. We can apply interceptors by decorating methods/classes with
+`@intercept`.
+
+## @intercept
+
+Syntax: `@intercept(...interceptorFunctionsOrBindingKeys)`
+
+The `@intercept` decorator adds interceptors to a class or its methods including
+static and instance methods. Two flavors are accepted:
+
+- An interceptor function
+- A binding key that can be resolved to an interface function
+
+### Method level interceptors
+
+A static or prototype method on a class can be decorated with `@intercept` to
+apply interceptors. For example,
+
+```ts
+class MyController {
+  @intercept(log)
+  static async greetStatic(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log)
+  static async greetStaticWithDI(@inject('name') name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(logSync)
+  greetSync(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log)
+  greet(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept('log')
+  async hello(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log)
+  async greetWithDI(@inject('name') name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept('log', logSync)
+  async helloWithTwoInterceptors(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  async helloWithoutInterceptors(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(changeName)
+  async helloWithChangeName(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(logWithErrorHandling)
+  async helloWithError(name: string) {
+    throw new Error('error: ' + name);
+  }
+}
+```
+
+### Class level interceptors
+
+To apply interceptors to be invoked for all methods on a class, we can use
+`@intercept` to decorate the class. When a method is invoked, class level
+interceptors (if not explicitly listed at method level) are invoked before
+method level ones.
+
+```ts
+@intercept(log)
+class MyControllerWithClassLevelInterceptors {
+  static async greetStatic(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log)
+  static async greetStaticWithDI(@inject('name') name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(log, logSync)
+  greetSync(name: string) {
+    return `Hello, ${name}`;
+  }
+
+  @intercept(convertName, log)
+  async greet(name: string) {
+    return `Hello, ${name}`;
+  }
+}
+```
+
+Here is the list of interceptors invoked for each method:
+
+- greetStatic: `log`
+- greetStaticWithDI: `log`
+- greetSync: `log`, `logSync`
+- greet: `convertName`, `log`
+
+## Invoke a method with interceptors
```

Comment:
In this section, how are we specifying the `interceptor` methods to use with `invokeMethod`?

Reply:
Controller methods decorated with `@intercept` are invoked with `invokeMethod` behind the scene.

Reply:
I added a sentence in the docs to make it clear.